File: client/src/app/shared/models/order.ts
export interface Order {
    id: number
    orderDate: string
    buyerEmail: string
    shippingAddress: ShippingAddress
    deliveryMethod: string
    shippingPrice: number
    orderItems: OrderItem[]
    subtotal: number
    status: string
    total: number
  }
  
  export interface ShippingAddress {
    name: string
    line1: string
    line2: any
    city: string
    huyen: string
    quan: string
  }
  
  export interface OrderItem {
    productId: number
    productName: string
    pictureUrl: string
    price: number
    quantity: number
  }

  export interface OrderToCreate{
    cartId: string;
    deliveryMethodId: number;
    shippingAddress: ShippingAddress;
  }

File: Core/Entities/OrderAggregate/Order.cs
using System;

namespace Core.Entities.OrderAggregate;

public class Order : BaseEntity
{
    public DateTime OrderDate { get; set; } = DateTime.UtcNow;
    public required string BuyerEmail { get; set; }
    public ShippingAddress ShippingAddress { get; set; } = null!;
    public DeliveryMethod DeliveryMethod { get; set; } = null!;
    public List<OrderItem> OrderItems { get; set; } = [];
    public decimal Subtotal { get; set; }
    public PaymentStatus Status { get; set; } = PaymentStatus.Pending;
    public decimal GetTotal()
    {
        return Subtotal + DeliveryMethod.Price;
    }
}
